### ISSUE TITLE
Fix for `"arrow-body-style": [1, "as-needed"]` causes other linter violations

### ISSUE THREAD

**Issue by scottyschup** [eslint/eslint#8548]
### Environment
* **ESLint Version:** 3.19.0
* **Node Version:** 6.5.0
* **Atom Version:** 1.16.0
* **steelbrain/linter Plugin Version:** 2.1.4
* **AtomLinter/linter-eslint Plugin Version:** 8.1.7

### Configuration
```
module.exports = {
    "extends": "airbnb-base",
    "plugins": [
        "import"
    ],
    "rules": {
      "arrow-body-style": [1, "as-needed"],
      "max-len": [2, 80, 2, { "ignoreComments": true }],
      "no-console": [1, {
        "allow": ["assert", "error", "info", "time", "warn"]
      }],
      "quote-props": [1, "consistent-as-needed"],
    }
};
```
### Problem
I have the "Fix errors on save" option turned on in Atom which applies ESLint fix upon saving. 

Given:
```js
// The code as I've typed it
const allUploadsDone = project.uploads.every((ul) => {
  return ul.status === 'completed' || ul.status === 'failed';
});
```
Converting the above code into a single-line expression would create a `max-len` violation, so I expect the line to pass `arrow-body-style` unaffected when the `as-needed` option is specified. However when I save, the block is converted into a single-line expression:
```js
// The single-line version that ESLint produces when I save in Atom
const allUploadsDone = project.uploads.every(ul => ul.status === 'completed' || ul.status === 'failed');
```
It seems like the use of `as-needed` should be mean that a fix is only applied if doing so does not introduce `max-len` (or any other) warnings/errors. Especially in this case since `max-len` violations are errors whereas `arrow-body-style` violations are only warnings.

**Comment by ilyavolodin** [comment]
This is expected behavior. `as-needed` option means that arrow functions should use short syntax when possible. It has nothing to do with other rules (in general, all of ESLint rules don't know about each other). `max-len` rule itself is not fixable (as in, auto-fix functionality doesn't support it), so it would not break up long lines automatically.

**Comment by not-an-aardvark** [comment]
Closing because this is working as intended.
